Asynchronous-communication

aki800 opened the pull request:
#WHAT
メッセージ投稿を非同期通信で行えるようにした。

#WHY
サーバーからのレスポンスが来なくとも通信が行えるようにするため。

↓メッセージを投稿したときに、メッセージ一覧部分が自動で最下部までスクロールされる様子
https://i.gyazo.com/a57dce3cd5d41cc53420930d9972c2ec.mp4

Commit:
Asynchronous-communication

#WHAT
メッセージ投稿を非同期通信で行えるようにした。

#WHY
サーバーからのレスポンスが来なくとも通信が行えるようにするため。

TOMARUMARU commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,45 @@
+  $(function(){
+    function buildHTML(message){
+      var addImage = (message.image.url !== null) ? `<img class="lower-message_image" src="${message.image.url}">"` : ''
+      console.log(message.image.url);
```
デバッグ消しましょう

aki800 replied:
console.log(message.image.url);を消しました。

TOMARUMARU commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,45 @@
+$(document).on('turbolinks:load', function() {
+
+  $(function(){
+    function buildHTML(message){
```
こちらのfunctionの中のインデント見直しましょう

aki800 replied:
インデントを見直しました。

TOMARUMARU commented on app/views/messages/create.json.jbuilder:
```diff
@@ -0,0 +1,6 @@
+json.body @message.body
+json.image @message.image
+json.time @message.created_at.strftime("%Y/%m/%d %H:%M")
+json.user_id @message.user.id
```
user_idは不要ですかね

aki800 replied:
json.user_id @message.user.idを消しました。

Commit:
Asynchronous-communication

#WHAT
デバッグとその他不要なものを削除
インデントの見直し
#WHY
コードを見やすくするため

hzak2xx commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,43 @@
+      .done(function(data){
+        var html = buildHTML(data);
+        $('.chat__user__messages').append(html)
+        $('.form__message').val('')
```
form属性のresetメソッドを使いましょう

aki800 replied:
valメソッドの代わりにresetメソッドを使用しました

Commit:
Use resetメソッド　in message.js

#WHAT
メッセージ投稿をしたときformから投稿内容を消すために、Resetメソッドを使用

#WHY
効率的にコードを書くため

TOMARUMARU commented on app/assets/javascripts/message.js:
```diff
@@ -31,7 +31,7 @@ $(document).on('turbolinks:load', function() {
         var html = buildHTML(data);
         $('.chat__user__messages').append(html)
-        $('.form__message').val('')
+        document.form.reset()
```
$('.form__message')に対してresetメソッドを使いましょう

aki800 replied:
formに設定されたidの値$('#new_message')に対してresetメソッドを使用しました。
($('.form__message')に対してだとresetメソッドがうまく働きませんでした。）

Commit:
Use reset メソッド　in message.js 2

#WHAT
Resetメソッドをformのid名（#new_message）に対して使用

#WHY
formの属性にNameを設定してresetメソッドを使用する必要がないため

TOMARUMARU reviewed (approved):
LGTM
